builder/internals.go

```diff
@@ -636,7 +644,11 @@ func (b *Builder) checkPathForAddition(orig string) error {
 		}
 		return err
 	}
-	if !strings.HasPrefix(origPath, b.contextPath) {
+	contextPath, err := filepath.EvalSymlinks(b.contextPath)
```

Comment:
It was a bug we found with long pathnames where the string comparison was failing as Windows was shortening the path to use the ~-style short name.